spk/shairport/src/dsm-control.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+
+# Package
+PACKAGE="shairport"
+DNAME="ShairPort"
+
+# Others
+SHAIRPORT="${SYNOPKG_PKGDEST}/bin/shairport"
+VAR_DIR="${SYNOPKG_PKGDEST}/var"
+PID_FILE="${VAR_DIR}/${PACKAGE}.pid"
+
+start_daemon ()
+{
+    # Launch the service in the background.
+    ${SHAIRPORT} --daemon --pidfile="$PID_FILE" --log=${VAR_DIR}/${PACKAGE}.log
+    # Wait until the service  is ready (race condition here).
+    counter=5
+    while [ $counter -gt 0 ]
+    do
+        daemon_status && break
+        let counter=counter-1
+        sleep 1
+    done
+}
+
+stop_daemon ()
+{
+    # TODO figure out a cleaner way to stop shairport
+    kill `cat ${PID_FILE}`
```

Comment:
With the `start-stop-daemon` approach mentioned in the previous comment, this would look nicer. Something like this: https://github.com/SynoCommunity/spksrc/blob/master/spk/tt-rss/src/dsm-control.sh#L25-L26